src/expression/expression_tree.rs

```diff
@@ -173,12 +173,20 @@ impl fmt::Display for Atom {
     }
 }
 
+type AlVec<T> = alloc::vec::Vec<T>;
+
 //Expression: a tree representing a mathematical expression
 #[derive(Debug, Clone, PartialEq)]
 pub enum Expression {
     //atoms
     Atom(Atom),
 
+    // Vector
+    Vector {
+        backing: AlVec<Expression>, // could use heapless::Vec instead of alloc::vec::Vec
+        size: u8,
```

Comment:
I'm a little unclear on why a size indicator is needed?

Reply:
I was hoping to have it match up with how matrices would be implemented down the line. I was thinking something along the lines of
```rust
Matrix {
  backing: Vec<Expression>,
  shape: (u8, u8),
}
```
so that the size of a vector is also its shape (either `(u8, 1)` or `(1, u8)`) depending on whether or not the vector should be treated as row-major or column-major. 

I thought it would be a more symmetric design but if you disagree then it can be removed as I'd imagine that getting the length of an `alloc::vec::Vec` and a `heapless::Vec` are both O(1).

Reply:
I mean, you're more of an expert on Vectors and Matrices than I am, so if it makes more sense to do that, I'll trust you on it.

Reply:
I'll have to sit on the design a little more. Still trying to navigate the codebase. Once the modifiers and evaluators start getting some hard code down, I'll see if this model fits better or not.

Reply:
For now, I don't think it would add that much overhead. It is only a `u8` after all. It is not like I'm storing a `usize`.